endgame_postprocessing/model_wrappers/sch/run_sch.py

```diff
@@ -47,19 +48,49 @@ def probability_any_worm(probability_for_each_worm: Iterable[float]):
     prob_not_any_worm = reduce(mul, prob_of_not_each_worm, 1.0)
     return 1.0 - prob_not_any_worm
 
+def probability_any_worm_max(probability_for_each_worm: Iterable[float]):
+    """
+    Calculate the probability of having any worm, given by the highest probability
+    among all the worms. Used for SCH.
 
-def canoncialise_single_result(file_info):
-    raw_iu = pd.read_csv(file_info.file_path)
-    raw_without_columns = raw_iu.drop(columns=["intensity", "species"])
-    # TODO: canonical shouldn't need the age_start / age_end but these are assumed present later
-    return canonicalise.canonicalise_raw(
-        raw_without_columns, file_info, "Prevalence SAC"
-    )
+    Inputs:
+     - probability_for_each_worm: Probability of having each worm
 
+    Returns: the probability of having any worm.
+    """
+    return reduce(lambda x, y: np.maximum(x, y), probability_for_each_worm)
```

Comment:
Wait, isn't that just `max(probability_for_each_worm)`? If yes I'd inline this function.

Reply:
I don't believe the built-in max function does element-wise comparison, so we need to use np.maximum

Reply:
Oh, I see. this functions returns an array instead of a single number.